runtime/src/accounts_db.rs

```diff
@@ -9676,7 +9676,9 @@ pub mod tests {
 
         // there has to be an existing append vec at this slot for a new current ancient at the slot to make sense
         let _existing_append_vec = db.create_and_insert_store(slot0, 1000, "test");
-        let _shrink_in_progress = current_ancient.create_ancient_append_vec(slot0, &db);
+        {
+            let _shrink_in_progress = current_ancient.create_ancient_append_vec(slot0, &db);
+        }
```

Comment:
I'm not always a huge fan of these guards. They're here to force a drop, but are one mental step removed (imo): scope ends -> drop. We could just explicitly drop it, in fewer lines, without the minor mental leap:
```suggestion
        let shrink_in_progress = current_ancient.create_ancient_append_vec(slot0, &db);
        drop(shrink_in_progress);
```